tests/e2e-tests/src/ai-core.test.ts

```diff
@@ -1,20 +1,141 @@
 import path from 'path';
 import { fileURLToPath } from 'url';
 import dotenv from 'dotenv';
-import { DeploymentApi } from '@sap-ai-sdk/ai-api';
-import 'dotenv/config';
+import {
+  AiDeployment,
+  AiDeploymentList,
+  DeploymentApi,
+  ScenarioApi
+} from '@sap-ai-sdk/ai-api';
+import _ from 'lodash';
 
 // Pick .env file from root directory
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 dotenv.config({ path: path.resolve(__dirname, '../.env') });
 
-describe('ai-api', () => {
-  it('should get deployments', async () => {
-    const deployments = await DeploymentApi.deploymentQuery(
-      {},
-      { 'AI-Resource-Group': 'default' }
-    ).execute();
-    expect(deployments).toBeDefined();
+describe('AI Core APIs', () => {
+  describe('DeploymentAPI', () => {
+    const MAX_RETRIES = 30;
+    const RETRY_INTERVAL = 5000;
+    let createdDeploymentId: string | undefined;
+    let initialState: AiDeploymentList | undefined;
+
+    async function waitForDeploymentToReachStatus(
+      deploymentId: string,
+      targetStatus: 'RUNNING' | 'STOPPED'
+    ): Promise<AiDeployment> {
+      for (let i = 0; i < MAX_RETRIES; i++) {
+        const deploymentDetail = await DeploymentApi.deploymentGet(
+          deploymentId,
+          {},
+          { 'AI-Resource-Group': 'i745181' }
+        ).execute();
+        if (deploymentDetail.status === targetStatus) {
+          return deploymentDetail;
+        }
+        await new Promise(resolve => setTimeout(resolve, RETRY_INTERVAL));
+      }
+      throw new Error(
+        `Deployment ${deploymentId} did not reach ${targetStatus} status within the expected time. ` +
+          'Please manually stop and delete the deployment.'
+      );
+    }
+
+    const sanitizedState = (state: AiDeploymentList | undefined) => ({
```

Comment:
[q] Why do we need this?

Reply:
For the pre and post condition check, I am currently checking if deploymentQuery (GET all deployments) is equal before and after the flow is executed, but it turns out that the 'modifiedAt' property is updated for all deployments on every get request. So I use this helper to remove this property while comparing the state.